### PR TITLE
contrib/internal/httptrace: collect client ip if DD_TRACE_CLIENT_IP_ENABLED is set to true

### PR DESCRIPTION
**Pull request by lievan**
### What does this PR do?
This PR enables the collection of client ip address if `DD_TRACE_CLIENT_IP_ENABLED` is set to `true`. IP related span tags are generated using the `ClientIPTags` function from `internal/appsec/dyngo/instrumentation/httpsec`

### Motivation
Users should be able to enable client ip collection even when ASM is not enabled. 

Also closes https://github.com/DataDog/dd-trace-go/issues/1553

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: contrib/internal/httptrace/httptrace.go [hunk 1]
```diff
@@ -11,15 +11,35 @@ import (
...
+// returns false when DD_TRACE_CLIENT_IP_ENABLED is unset, set to false, or cannot be parsed to a boolean value
+func collectClientIP() (enabled bool, err error) {
+	if enabledStr, set := os.LookupEnv(envTraceClientIPEnabled); !set {
+		return false, nil
+	} else if enabled, err = strconv.ParseBool(enabledStr); err != nil {
+		return false, fmt.Errorf("could not parse %s value `%s` as a boolean value", envTraceClientIPEnabled, enabledStr)
+	} else {
+		return enabled, nil
+	}
+}
```
Comment (felixge) [inline contrib/internal/httptrace/httptrace.go:42]: NIT: This looks a lot like `internal.BoolEnv`, maybe you could (re)use that instead?

https://github.com/DataDog/dd-trace-go/blob/74a8e95a31d9fc95d2f1a7b804fabe7f2467cb0c/internal/env.go#L16-L29

File: contrib/internal/httptrace/httptrace.go [hunk 2]
```diff
@@ -41,6 +61,14 @@ func StartRequestSpan(r *http.Request, opts ...ddtrace.StartSpanOption) (tracer.
 	if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(r.Header)); err == nil {
 		opts = append(opts, tracer.ChildOf(spanctx))
 	}
+	if enabled, err := collectClientIP(); err != nil {
```
Comment (felixge) [inline contrib/internal/httptrace/httptrace.go:64]: NIT: I think the usual pattern is to lookup and parse env based config vars in the config and to just check the config in places like this. That keeps the code a little more organized and also has performance benefits.

https://github.com/DataDog/dd-trace-go/blob/74a8e95a31d9fc95d2f1a7b804fabe7f2467cb0c/contrib/internal/httptrace/config.go#L28-L49

File: contrib/internal/httptrace/httptrace_test.go [hunk 3]
```diff
@@ -24,11 +30,124 @@ func TestStartRequestSpan(t *testing.T) {
...
-
 	require.Len(t, spans, 1)
 	assert.Equal(t, "example.com", spans[0].Tag("http.host"))
 }
 
+// a test logger for TestClientIP
+type testLogger struct {
+	mu    sync.RWMutex
+	lines []string
+}
```
Comment (felixge) [inline contrib/internal/httptrace/httptrace_test.go:41]: NIT: Could you reuse the RecordLogger from there? It doesn't seem to have a Reset method right now, but that could probably be added.

https://github.com/DataDog/dd-trace-go/blob/74a8e95a31d9fc95d2f1a7b804fabe7f2467cb0c/internal/log/log.go#L200-L220

**Review by felixge — COMMENTED** [review]
Thanks @lievan 🥳 . I just had a quick look and added some suggestions.

File: contrib/internal/httptrace/httptrace.go (hunk 2 above)
Comment (Julio-Guerra) [reply contrib/internal/httptrace/httptrace.go:64]: Indeed, it is important doing the initializations and configurations out of the "hot path" when possible.

File: contrib/internal/httptrace/httptrace_test.go [hunk 4]
```diff
@@ -9,12 +9,18 @@ import (
...
+	"strconv"
+	"sync"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo/instrumentation"
```
Comment (Julio-Guerra) [inline contrib/internal/httptrace/httptrace_test.go:22]: A shame this one "leaked out" but I think we should have kept those IP-related functions in `httpsec` in the end, which would have been cleaner to use for you here. We plan to do this small refactoring soon, so we'll adjust this file too when we do.

**Review by katiehockman — DISMISSED** [review]
Looks good! Just a few comments about the tests. Thanks!

**Review by felixge — APPROVED** [review]
LGTM. I reviewed the changes made in response to my comments and checked for general issues in the code. I didn't review the use case and test cases in detail, but @katiehockman's comments make sense.

**Review by Julio-Guerra — APPROVED** [review]

**Review by katiehockman — APPROVED** [review]

**Pull request merged**